### ISSUE TITLE
Improve Sample app

### ISSUE THREAD

**Issue by sanogueralorenzo** [NYTimes/Store#273]
Some suggestions, I would be happy to help in some of them if needed:

1. Sample app has currently 2 activities and one of them is completely unused.

2. Comments in several parts of the App would be appreciated.

3. You need to check the readme while looking the Sample app to figure things out and even when doing so readme has things like getArticleAsBufferedSource but nothing like that can't  be found in the sample app.

4. Does the API persister only accepts the response body?  Making the need to have 2 interfaces per api call?

5. Could a sample project in Kotlin be provided?

6. Is the persister required for disk caching? And do we have only 1 accepting response body so we can have just 1 for all of the different stores?

7. Is the provideRedditStore only caching in memory? Would a better name be suitable?

8. Sample app is using some deprecated methods from MemoryPolicy (setExpiredAfter)
 
I'm trying to build a kotlin one and write a medium story about it but I'm facing too many doubts to feel comfortable doing so or contributing.

**Comment by digitalbuddha** [comment]
Sorry for the terrible state of the sample it never received attention went we released store3.

I will fix 1,2,3 as for 4 I'm not sure what you mean, sorry mind explaining in more detail?

5, at the bottom of the Readme there is a community project in kotlin. Also the sample cat app and 360andev talk near bottom is a kotlin project.

6. Yes persister is the disk caching

7. Yes

8 need to be fixed


We obviously would take external contributions to clean stuff up. Otherwise do when I have some free time.

Also the  android makers and droidcon Italy talks on bottom explain store in the simplest way.

**Comment by sanogueralorenzo** [comment]
Hello and thanks for the fast response!

No worries! Hope my suggestions and myself can contribute to the Sample project of Store.

4 and 6 are kind of related. I will try with 4 again: Is only 1 persister required for all of the stores? Does the persister just stores in cache the response bodies of all of the different apis (with keys) so when you want to get something from cache you are just parsing the response body again from cache into an object?

Thanks for 5, missed that I will have a look straight away and also to the android makers and droidcon Italy things.

Once I'm more familiar I will open up some PRs if that is okay.

Will write an entry medium post I will run it over here in case you want to have a look.

Thanks

**Comment by digitalbuddha** [comment]
Yup you can use the same `Persister` implementation across multiple stores.  If you use our `FileSystem` based persister implementations it does in fact store the raw response json on disk.  You can also build your own persister as well that can work with any input type and output type you see fit. I've seen apps that use Retrofit, no parser and their own   `SqlPersister` which works on java objects rather than raw values from network.  

Anyways lmk how I can help and good luck figuring things out (The tests are pretty useful to see how things work)

**Comment by KevCron** [comment]
1. I believe the intention behind only using one of the activities comes from the fact that `StoreActivity` is supposed to demonstrate using the `nonPersistedStore` while the `PersistedStoreActivity` is supposed to demonstrate using the `persistedStore`.

2. I've added comments where I thought was appropriate, mostly around Store related behavior. @sanogueralorenzo Is there anything else you think needs comments?

3. I think the comments I've added should help with this. The `getArticleAsBufferedSource` you mentioned isn't specific to Store itself, that's the just an example api that might be used by a user of Store. The equivalent in the sample app would be `fetchSubreddit`. If you still think that's unclear I can add some more comments explaining within the sample app.

4. Addressed by Mike.

5. If the community project is insufficient we can raise a new issue for creating a new sample app entirely in Kotlin. Perhaps this would be a good task for the Store 4 redesign?

6. Addressed by Mike.

7. Addressed by Mike.

8. Seems to have been fixed, can't see any usages of `setExpiredAfter` within the sample app. Can only find `setExpireAfterWrite` and `setExpireAfterTimeUnit`.

Relevant PR: https://github.com/NYTimes/Store/pull/335
